adoption/src/main/java/org/dcsa/conformance/standards/adoption/action/SupplyScenarioParametersAction.java

```diff
@@ -0,0 +1,104 @@
+package org.dcsa.conformance.standards.adoption.action;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.Getter;
+import org.dcsa.conformance.core.scenario.ConformanceAction;
+import org.dcsa.conformance.standards.adoption.party.FilterParameter;
+import org.dcsa.conformance.standards.adoption.party.SuppliedScenarioParameters;
+
+@Getter
+public class SupplyScenarioParametersAction extends ConformanceAction {
+  public static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+  private SuppliedScenarioParameters suppliedScenarioParameters = null;
+  private final LinkedHashSet<FilterParameter> filterParameters;
+
+  public SupplyScenarioParametersAction(
+      String publisherPartyName, FilterParameter... filterParameters) {
+    super(
+        publisherPartyName,
+        null,
+        null,
+        "SupplyScenarioParameters(%s)"
+            .formatted(
+                Arrays.stream(filterParameters)
+                    .map(FilterParameter::getQueryParamName)
+                    .collect(Collectors.joining(", "))));
+    this.filterParameters =
+        Stream.of(filterParameters).collect(Collectors.toCollection(LinkedHashSet::new));
```

Comment:
Could this be simplified to `new LinkedHashSet<>(Arrays.asList(filterParameters))`?

Reply:
Yes indeed. This class I didn't give much attention actually.